packages/kbn-ftr-common-functional-services/services/retry/retry.ts

```diff
@@ -10,19 +10,33 @@ import { FtrService } from '../ftr_provider_context';
 import { retryForSuccess } from './retry_for_success';
 import { retryForTruthy } from './retry_for_truthy';
 
+interface TryWithRetriesOptions {
+  retryCount: number;
+  retryDelay?: number;
+  timeout?: number;
+}
+
 export class RetryService extends FtrService {
   private readonly config = this.ctx.getService('config');
   private readonly log = this.ctx.getService('log');
 
+  /**
+   * Use to retry block within {timeout} period and return block result.
+   * @param timeout retrying timeout
+   * @param block retrying operation
+   * @param onFailureBlock optional block to run on operation failure before the new attempt
+   * @param retryDelay optional delay before the new attempt
+   * @returns result from block
+   */
   public async tryForTime<T>(
     timeout: number,
     block: () => Promise<T>,
     onFailureBlock?: () => Promise<T>,
     retryDelay?: number
   ) {
-    return await retryForSuccess(this.log, {
+    return await retryForSuccess<T>(this.log, {
       timeout,
-      methodName: 'retry.tryForTime',
+      methodName: 'retry.retryWithRetries',
```

Comment:
I don't really get why the method name has been changed here. Could you please explain that a bit?